### ISSUE TITLE
startingDirectory not honored while distro is in stopped state

### ISSUE THREAD

**Issue by CoolCat64** [microsoft/terminal#8251]
# Environment

Windows build number: Microsoft Windows [Version 10.0.19042.630]
Windows Terminal version (if applicable): 1.3.2651.0
uname -mrs: Linux 4.19.128-microsoft-standard x86_64
lsb_release -a:
No LSB modules are available.
Distributor ID: Kali
Description:    Kali GNU/Linux Rolling
Release:        2020.4
Codename:       kali-rolling

Any other software?
I have installed Ubuntu-20.04 and kali-linux from the Windows Store.

# Steps to reproduce
0.) Use the config files included in this PR
1.) Set the settings.json kali-linux entry with startingDirectory to any of the following:
//wsl$/kali-linux/home/\<username\>
\\\\wsl$\\kali-linux\\home\\\<username\>
2.) Verify kali-linux is in a "Stopped" state via wsl -l -v
3.) Open a new Windows Terminal tab with kali-linux

# Expected behavior
I expect the starting directory to be //wsl$/kali-linux/home/\<username\> per documentation

https://docs.microsoft.com/en-us/windows/terminal/customize-settings/profile-settings#starting-directory

# Actual behavior
The starting directory is the default %USERPROFILE%.
When, according to wsl -l -v, the kali-linux instance is already 'Running', then opening a tab for kali-linux does set the directory to //wsl$/kali-linux/home/\<username\> as expected.

I tried enabling the automount option, I tried setting the mountFsTab to false, and commenting out all automount options. The behavior is the same, so long as the kali-linux instance is stopped before opening the tab.

[config_files.zip](https://github.com/microsoft/terminal/files/5533251/config_files.zip)

**Comment by zadjii-msft** [comment]
Thanks for the great writeup! I think largely, we're tracking this issue over at #7197 - feel free to follow that discussion for updates on this bug. 

/dup #7197

**Comment by CoolCat64** [comment]
I don't get it. I can no longer reproduce the problem and I am not sure why. I updated some files, clone some repos, and ran rsync a couple of times. The most destructive thing I did was run the optimize-vhd cmdlet (that is a Win10 Pro Hyper-V cmdlet for optimizing virtual disk drives).

The one thing I did notice was that both Ubuntu-20.04 and kali-linux would run when I opened a kali-linux tab, where before, Ubuntu-20.04 was "Stopped". I am assuming this has something to do with a mount command in my .bashrc:

`# Mount the Ubuntu filesystem to the wsl share directory (tmpfs is lost every reboot)
if [ ! -d /mnt/wsl/Ubuntu-20.04 ]; then
   mkdir /mnt/wsl/Ubuntu-20.04
   wsl.exe -d Ubuntu-20.04 -u root mount --bind / /mnt/wsl/Ubuntu-20.04
fi`

But this mount command is in the original config file set that I submitted with this PR. When the .bashrc was being sourced before, this was being run when I opened an instance of kali-linux, but the Ubuntu-20.04 in the wsl -l -v output would still show "Stopped"...I am very confused.
